### PR TITLE
Enforce pylint conventions

### PR DESCRIPTION
**Pull request by EmilyBourne**
Add an additional test to complain when unexpected pylint disable commands are added. This test recommends the use of pylint disable commands for docstrings in the tests. Additionally the disabling of docstrings for `_print_X` and `_visit_X` functions is now handled in the .pylintrc file. This means that this only applies to those functions instead of applying to all the functions in the file.

### CODE REVIEW

**Review by yguclu — CHANGES_REQUESTED** [review]
Very nice!

File: ci_tools/check_pylint_commands.py [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+        disabled = []
+        for l in pylint_lines:
+            disabled.extend(l.split('=')[1].split(','))
+        for r,d in accepted_pylint_commands.items():
+            if r.match(f):
+                for di in d:
+                    try:
+                        disabled.remove(di)
+                    except ValueError:
+                        pass
```
[the remark this answers is not in the document]
Comment (EmilyBourne) [reply ci_tools/check_pylint_commands.py]: Fixed [3fb10b6](https://github.com/pyccel/pyccel/pull/1359/commits/3fb10b66356c4f2a5a113fd4eb723c0d542a59b1) (and [2f37bd2](https://github.com/pyccel/pyccel/pull/1359/commits/2f37bd200df46b176734aec33d597478eb2b6fd1))

Using a try/except tends to be faster on Python which is why I wrote it this way, but it seems that sets have built-in ways to manage this

[the remark this answers is not in the document]
Comment (yguclu) [reply ci_tools/check_pylint_commands.py]: Indeed working with sets instead of lists makes a lot of sense here!

**Review by yguclu — APPROVED** [review]

**Pull request merged**